packages/sdk/clients/decorators/erc7579/index.ts

```diff
@@ -0,0 +1,148 @@
+  ) => Promise<Hash>
+  isModuleInstalled: (
+    args: IsModuleInstalledParameters<TSmartAccount>
+  ) => Promise<boolean>
+  supportsExecutionMode: (
+    args: SupportsExecutionModeParameters<TSmartAccount>
+  ) => Promise<boolean>
+  supportsModule: (
+    args: SupportsModuleParameters<TSmartAccount>
+  ) => Promise<boolean>
+  uninstallModule: (
+    args: UninstallModuleParameters<TSmartAccount>
+  ) => Promise<Hash>
+  uninstallModules: (
+    args: UninstallModulesParameters<TSmartAccount>
+  ) => Promise<Hash>
+  getInstalledValidators: (
+    args: GetInstalledValidatorsParameters<TSmartAccount>
+  ) => Promise<readonly [readonly Hex[], Hex]>
+  getInstalledExecutors: (
+    args: GetInstalledExecutorsParameters<TSmartAccount>
+  ) => Promise<readonly [readonly Hex[], Hex]>
+  getActiveHook: (args: GetActiveHookParameters<TSmartAccount>) => Promise<Hex>
+  getFallbackBySelector: (
+    args: GetFallbackBySelectorParameters<TSmartAccount>
+  ) => Promise<[Hex, Hex]>
+}
+
+export type {
+  InstallModuleParameters,
+  IsModuleInstalledParameters,
+  CallType,
+  ExecutionMode,
+  SupportsExecutionModeParameters,
+  ModuleType,
+  SupportsModuleParameters,
+  UninstallModuleParameters,
+  GetInstalledValidatorsParameters,
+  GetInstalledExecutorsParameters,
+  GetActiveHookParameters
+}
+
+export {
+  accountId,
+  installModule,
+  installModules,
+  isModuleInstalled,
+  supportsExecutionMode,
+  supportsModule,
+  uninstallModule,
+  uninstallModules,
+  getInstalledValidators,
+  getInstalledExecutors,
+  getActiveHook,
+  getFallbackBySelector
+}
+
+export function erc7579Actions() {
+  return <TSmartAccount extends SmartAccount | undefined>(
+    client: Client<Transport, Chain | undefined, TSmartAccount>
+  ): Erc7579Actions<TSmartAccount> => ({
+    accountId: (args) => accountId(client, args),
+    installModule: (args) => installModule(client, args),
+    installModules: (args) => installModules(client, args),
+    isModuleInstalled: (args) => isModuleInstalled(client, args),
+    supportsExecutionMode: (args) => supportsExecutionMode(client, args),
+    supportsModule: (args) => supportsModule(client, args),
+    uninstallModule: (args) => uninstallModule(client, args),
+    uninstallModules: (args) => uninstallModules(client, args),
+    getInstalledValidators: (args) => getInstalledValidators(client, args),
+    getInstalledExecutors: (args) => getInstalledExecutors(client, args),
+    getActiveHook: (args) => getActiveHook(client, args),
+    getFallbackBySelector: (args) => getFallbackBySelector(client, args)
+  })
+}
+
+export type Module = {
+  address: Address
+  context: Hex
+  additionalContext?: Hex
+  type: ModuleType
+
+  /* ---- kernel module params ---- */
+  // these param needed for installing validator, executor, fallback handler
+  hook?: Address
+  /* ---- end kernel module params ---- */
+
+  /* ---- safe module params ---- */
+  // these two params needed for installing hooks
+  hookType?: SafeHookType
```

Comment:
not sure about the use of this.

Reply:
Me either... @VGabriel45 this was your addition too. Mind taking this?